### PR TITLE
Feature/update reviewer trailer

### PR DESCRIPTION
**Pull request by SindhuraGowri**
review request not fetching from commithash, instead validate it when user has given url in commit message

### CODE REVIEW

File: marge/job.py [hunk 1]
```diff
@@ -154,26 +154,37 @@ def add_trailers(self, merge_request):
         log.info('commitList %s', commitList)
         last_commit_json = commitList[0]
 
-        if (re.search(r'Review URL:\s*trivial\s*[\n\r\t]{0}', last_commit_json['message'], re.IGNORECASE)):
+        if (re.search(r'Review URL:\s*(trivial|none|auto)[\s]{0,}', last_commit_json['message'], re.IGNORECASE)):
```
Comment (RajasekharReddyY) [inline marge/job.py:157]: can you replace `Review URL:\s*(trivial|none|auto)[\s]{0,}`  with `Review URL:\s*(trivial|none|auto)\s*`

File: marge/job.py [hunk 2]
```diff
@@ -154,26 +154,37 @@ def add_trailers(self, merge_request):
...
-            )
-    
-            log.info(type(rbt_reviewers))
-            log.info('rbt approver %s', rbt_reviewers)
+            if not (re.search(r'Review URL:', last_commit_json['message'], re.IGNORECASE)):
+                rurl = 'none'
+                rbt_reviewers = 'none' if not (re.search(r'Reviewed by:', last_commit_json['message'], re.IGNORECASE)) else None
+
+            else:
+                rv_url = re.search(r"Review URL:\s*http[s]?://reviewboard.eng.vmware.com/r/(\w+).*",last_commit_json['message'],flags=re.IGNORECASE)
```
Comment (RajasekharReddyY) [inline marge/job.py:170]: rename `rv_url` to `review_url_match`

File: marge/job.py [hunk 3]
```diff
@@ -154,26 +154,37 @@ def add_trailers(self, merge_request):
...
+                if rv_url:
+                    review_id = rv_url.group(1).lstrip().rstrip()
+                    log.info(rv_url)
+                    log.info(review_id)
```
Comment (RajasekharReddyY) [inline marge/job.py:174]: log.info('Review ID: %s', review_id)

File: marge/job.py [hunk 4]
```diff
@@ -534,10 +547,13 @@ def _get_review_and_appr(status, user, commit):
...
-        requests = root.get_review_requests(status=status,user=user,commit_id=commit);
-        RURL = requests[0].absolute_url;
-        log.info('RURL: %s', RURL)
-        rr = root.get_reviews(review_request_id=requests[0].id)
+        # requests = root.get_review_requests(status=status,user=user,commit_id=commit);
+        # RURL = requests[0].absolute_url;
+        # log.info('RURL: %s', RURL)
+        # rr = root.get_reviews(review_request_id=requests[0].id)
+        log.info(review_id)
+        RURL = 'https://reviewboard.eng.vmware.com/r/'+review_id
```
Comment (RajasekharReddyY) [inline marge/job.py:555]: change `RURL` to `review_url`

File: marge/job.py [hunk 5]
```diff
@@ -534,10 +547,13 @@ def _get_review_and_appr(status, user, commit):
...
+        rr = root.get_reviews(review_request_id=review_id)
```
Comment (RajasekharReddyY) [inline marge/job.py:556]: rename `rr` to  `rbt_reviewers`

File: marge/job.py [hunk 6]
```diff
@@ -534,10 +547,13 @@ def _get_review_and_appr(status, user, commit):
...
         rbt_reviewers = [];
```
Comment (RajasekharReddyY) [inline marge/job.py:557]: rename `rbt_reviewers ` to `rbt_approvers`

File: marge/job.py [hunk 7]
```diff
@@ -534,10 +547,13 @@ def _get_review_and_appr(status, user, commit):
...
     try:
-        requests = root.get_review_requests(status=status,user=user,commit_id=commit);
-        RURL = requests[0].absolute_url;
-        log.info('RURL: %s', RURL)
-        rr = root.get_reviews(review_request_id=requests[0].id)
+        # requests = root.get_review_requests(status=status,user=user,commit_id=commit);
+        # RURL = requests[0].absolute_url;
+        # log.info('RURL: %s', RURL)
+        # rr = root.get_reviews(review_request_id=requests[0].id)
+        log.info(review_id)
```
Comment (RajasekharReddyY) [inline marge/job.py:554]: log.info('Review id %s', review_id)

File: marge/job.py [hunk 8]
```diff
@@ -154,26 +154,37 @@ def add_trailers(self, merge_request):
         log.info('commitList %s', commitList)
         last_commit_json = commitList[0]
 
-        if (re.search(r'Review URL:\s*trivial\s*[\n\r\t]{0}', last_commit_json['message'], re.IGNORECASE)):
+        if (re.search(r'Review URL:\s*(trivial|none|auto)[\s]{0,}', last_commit_json['message'], re.IGNORECASE)):
             log.info('trivial commit - do not query for review request')
-            #approvers_response_reviewer = merge_request.fetch_approvals()
-            #rbt_reviewers_updated = approvers_response_reviewer.approver_usernames
             (rbt_reviewers_updated, rurl_updated) = (None, None)
+            log.info('rbt_reviewers and rurl ---')
```
Comment (RajasekharReddyY) [inline marge/job.py:160]: change logging, prefix relevant string

File: marge/job.py (hunk 1 above)
Comment (SindhuraGowri) [reply marge/job.py]: replaced it

File: marge/job.py (hunk 8 above)
Comment (SindhuraGowri) [reply marge/job.py]: updated all the variables and logging...

**Pull request merged**